aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,1360 @@
+                let res_signer = create_signer_with_capability(&dao.dao_signer_capability);
+                let dst_addr = property_map::read_address(args, &string::utf8(b"dst"));
+                let amount = property_map::read_u64(args, &string::utf8(b"amount"));
+                transfer_fund(&res_signer, dst_addr, amount);
+            } else if (function_name == &string::utf8(b"offer_nft")) {
+                let res_signer = create_signer_with_capability(&dao.dao_signer_capability);
+                let creator = property_map::read_address(args, &string::utf8(b"creator"));
+                let collection = property_map::read_string(args, &string::utf8(b"collection"));
+                let token_name = property_map::read_string(args, &string::utf8(b"token_name"));
+                let property_version = property_map::read_u64(args, &string::utf8(b"property_version"));
+                let dst = property_map::read_address(args, &string::utf8(b"dst"));
+                offer_nft(&res_signer, creator, collection, token_name, property_version, dst);
+            } else {
+                assert!(function_name == &string::utf8(b"no_op"), error::invalid_argument(ENOT_SUPPROTED_FUNCTION));
+            };
+            i = i + 1;
+        };
+    }
+
+    /// Resolve an proposal
+    fun resolve_internal(resolver: Option<address>, proposal_id: u64, nft_dao: address) acquires DAO, Proposals, ProposalVotingStatistics {
+        // validate if proposal is ready to resolve
+        let dao = borrow_global_mut<DAO>(nft_dao);
+        // assert the proposal voting ended
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+
+        if (option::is_some(&resolver)) {
+            // only DAO admin can execute the proposal directly
+            assert!(*option::borrow(&resolver) == dao.admin, error::permission_denied(EINVALID_ADMIN_ACCOUNT));
+            execute_proposal(proposal, dao);
+            proposal.resolution = PROPOSAL_RESOLVED_BY_ADMIN;
+            // return early befor emitting the normal resolve event.
+            return
+        };
+
+        assert!(exists<ProposalVotingStatistics>(nft_dao), error::not_found(EVOTING_STATISTICS_NOT_FOUND));
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        proposal.final_yes_votes = voting_stat.total_yes;
+        proposal.final_no_votes = voting_stat.total_no;
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        } else if(voting_stat.total_yes > voting_stat.total_no) {
+            execute_proposal(proposal, dao);
```

Comment:
If `execute_proposal` failed (eg. no balance to transfer), the propsal will be PENDING. How to handle it? ( just `admin_veto` or `admin_resolve` ？)

Reply:
It could be 
1. execute the proposal again when there is sufficient fund
2. admin veto the proposal to remove the proposal

Reply:
no, when there is sufficient fund, anyone can resolve the proposal.